src/types.ts

```diff
@@ -0,0 +1,12 @@
+export type Book = {
+    id: string,
+    title: string
+    readings: Reading[]
+}
+
+export type Reading = {
+    id: string,
+    startDate: Date,
+    startPage: number,
+    finalDetails: null | {endDate: Date, endPage: number}
```

Comment:
```suggestion
    finalDetails?: {endDate: Date, endPage: number}
```
Nullable typy raczej w ten sposób się zapisuje + moża wydzielić obiekt do osobnego typu